### PR TITLE
Fixes bug with date serialisation, and git version change from "unstable" to "alpha"

### PR DESCRIPTION
**Pull request by dazinator**
When calling the jira rest API to get project versions, the ProjectVersion dto that gets deserialised from the json has acouple of dates that weren;t being handled correctly and were generating serialisation errors. This wasn't noticed previously as we didn't have these fields populated in Jira until actual issuing a versions as "Released" and entering in a release date.

Also we updated our version of GitVersion that we use during builds. The older GitVersion used to use "alpha" in the version numbers pre-release label, but now that has switched to "unstable". The MergeVersions functionality was explicitly looking for versions with "unstable" in the label to merge, but with this PR it now handles them both. In future I think we need to make this configurable.

### CODE REVIEW

File: src/JiraCli/Extensions/JiraExtensions.version.cs [hunk 1]
```diff
@@ -16,6 +16,15 @@ namespace JiraCli
     using Newtonsoft.Json.Linq;
     using RestSharp;
     using RestSharp.Serializers;
+    using Newtonsoft.Json.Converters;
+
+    class CustomDateTimeConverter : IsoDateTimeConverter
```
Comment (GeertvanHorrik) [inline src/JiraCli/Extensions/JiraExtensions.version.cs]: Move to separate class please

File: src/JiraCli/Services/Interfaces/IVersionInfoService.cs [hunk 2]
```diff
@@ -5,10 +5,14 @@
 // --------------------------------------------------------------------------------------------------------------------
 
 
+using System;
```
Comment (GeertvanHorrik) [inline src/JiraCli/Services/Interfaces/IVersionInfoService.cs]: Usings inside namespace

File: src/JiraCli/Services/VersionInfoService.cs [hunk 3]
```diff
@@ -95,6 +95,40 @@ private VersionComparisonResult ConvertToComparisonResult(int result)
...
+
+        public bool IsPreRelease(string version)
+        {
+            Argument.IsNotNull(() => version);            
+
+            // Can assume semver format.
+            SemVersion semVer;
+            if (SemVersion.TryParse(version, out semVer))
+            {
+                if (!string.IsNullOrWhiteSpace(semVer.Prerelease))
```
Comment (GeertvanHorrik) [inline src/JiraCli/Services/VersionInfoService.cs:123]: Note sure what the point of this code is. Does this method ever return true?

**Review by GeertvanHorrik — CHANGES_REQUESTED** [review]
Overall great pr, thanks for the fixes. A few small things (see comments). Next time also please don't commit changes to files that aren't really relevant for the change you are proposing (such as sln, packages.config and FodyWeavers).

**Comment by dazinator** [comment]
> Next time also please don't commit changes to files that aren't really relevant for the change you are proposing (such as sln, packages.config and FodyWeavers).

Yep - sorry, my bad.. it's just because in my fork I ended up adding this as a submodule and building as part of another solution whilst debugging the issue. I'll be more careful next time.

File: src/JiraCli/Services/VersionInfoService.cs (hunk 3 above)
Comment (dazinator) [reply src/JiraCli/Services/VersionInfoService.cs:123]: Yeah it does.. It returns true if the version has a Prerelease label, and the predicate check against that label returns true. The predicate we pass in, checks that the label starts with "alpha" or "unstable". 

It allows the caller to make an additional assertion against the pre-release label portion of a pre-release version number when the version number passed in is actually a pre-release version.

Comment (GeertvanHorrik) [reply src/JiraCli/Services/VersionInfoService.cs:123]: Now I see the return prereleaseLabelChecker. I swear yesterday it had an empty { } or I must have been dreaming.

Comment (GeertvanHorrik) [reply src/JiraCli/Services/VersionInfoService.cs:123]: I double checked, I commented on the wrong line. See IsPreRelease(stringVersion) instead.

Comment (dazinator) [reply src/JiraCli/Services/VersionInfoService.cs:123]: Haha yep. Will tidy up 😂

**Comment by GeertvanHorrik** [comment]
@dazinator did you find time to tidy it up or is it already ready to be merged?

**Comment by dazinator** [comment]
Afraid I haven't had a chance to tidy this up just yet.

**Comment by dazinator** [comment]
I have adjusted as per the feedback. Should be ready to merge.

**Comment by GeertvanHorrik** [comment]
Excellent, thanks!

**Review by GeertvanHorrik — APPROVED** [review]

**Comment by GeertvanHorrik** [comment]
New version will be available in a few minutes.

**Comment by dazinator** [comment]
Sadly I think the build might fail, apologies. Have submitted a missed commit.

**Comment by GeertvanHorrik** [comment]
Already fixed it ;-)

**Comment by dazinator** [comment]
Cheers

**Pull request merged**